extensions/notification/NotificationQueuePopulator.js

```diff
@@ -119,22 +119,78 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
         return this.bnConfigManager.removeConfig(bucketName);
     }
 
+    /**
+     * Returns the correct versionId
+     * to display according to the
+     * versioning state of the object
+     * @param {Object} value log entry object
+     * @param {Object} overheadFields - extra fields missing from the log entry
+     * @return {String} versionId
+     */
+    _getVersionId(value, overheadFields) {
+        const versionId = value.versionId || (overheadFields && overheadFields.versionId);
+        const isNullVersion = value.isNull;
+        const isVersioned = !!versionId;
+        // Versioning suspended objects have
+        // a versionId, however it is internal
+        // and should not be used to get the object
+        if (isNullVersion || !isVersioned) {
+            return null;
+        }
+        return versionId;
+    }
+
+    /**
+     * Decides if we should process the entry.
+     * Since we get both master and version events,
+     * we need to avoid pushing two notifications for
+     * the same event.
+     * - For non versiond buckets, we process the master
+     * objects' events.
+     * - For versioned buckets, we process version events
+     * and ignore all master events.
+     * - For versioning suspended buckets, we need to process
+     * both master and version events, as the master is considered
+     * a separate version.
+     * @param {String} key object key
+     * @param {Object} value object metadata
+     * @return {boolean} - true if entry is valid
+     */
+    _shouldProcessEntry(key, value) {
+        const isMaster = isMasterKey(key);
+        const isVersionedMaster = isMaster && !!value.versionId;
+        const isNullVersion = isMaster && value.isNull;
+        if (!isMaster
+            || !isVersionedMaster
+            || isNullVersion
+            ) {
+            return true;
+        }
+        return false;
```

Comment:
Style suggestion, but should be reworked due to the above comment.

```suggestion
    return (!isMaster
        || !isVersionedMaster
        || isNullVersion
        );
```

Reply:
I was trying to keep the code and jsdoc the same as in backbeat 8.x where it comes from, to ease forward porting and eventual use of backbeat 8.x in RING, but given the discrepancy you pointed out between Artesca and RING it's not much use anymore.